packages/spatial/src/physics/classes/Physics.ts

```diff
@@ -302,11 +307,22 @@ const setRigidBodyType = (world: PhysicsWorld, entity: Entity, type: Body) => {
       break
   }
 
-  rigidbody.setBodyType(typeEnum, false)
+  rigidbody.setBodyType(typeEnum, true)
 
   /** @todo turns out this is a react/rapier bug when it comes to changing the rigidbody type. This is the best workaround I could find*/
   rigidbody.setEnabled(false)
   rigidbody.setEnabled(true)
+
+  const childColliderEntities = getChildrenWithComponents(entity, [ColliderComponent])
+  const colliderEntities = hasComponent(entity, ColliderComponent)
+    ? [entity, ...childColliderEntities]
+    : childColliderEntities
+
+  const nestedCollidersState = getMutableState(NestedCollidersState)
```

Comment:
it makes me nervous to rely on state like this in the physics abstraction. This means that the physics API and the component itself is handling data that feeds back on itself. we need a uni-directional flow of responsibilities here.

The Physics API exists purely to translate component data into the physics world, not to manipulate the state of the ECS.